dsfhub/config.go

```diff
@@ -22,17 +22,26 @@ type Config struct {
 
 var missingAPITokenMessage = "DSF HUB API Token must be provided"
 var missingDSFHostMessage = "DSF HUB host/API endpoint must be provided"
+var validSyncTypes = []string { "SYNC_GW_BLOCKING", "SYNC_GW_NON_BLOCKING", "DO_NOT_SYNC_GW"}
+var invalidSyncTypeMessage = "Invalid sync_type. Available values: " + strings.Join(validSyncTypes, ", ")
 
 // Client configures and returns a fully initialized DSF Client
 func (c *Config) Client() (interface{}, error) {
-	// Check DSFToken env var
+	// Check DSFToken 
 	if strings.TrimSpace(c.DSFHUBToken) == "" {
 		return nil, errors.New(missingAPITokenMessage)
 	}
-	// Check DSFHost env var
+	// Check DSFHost 
 	if strings.TrimSpace(c.DSFHUBHost) == "" {
 		return nil, errors.New(missingDSFHostMessage)
 	}
+	// Check sync_type param
+	syncType, exists := c.Params["syncType"]
+	if exists {
```

Comment:
Not necessary at all, but if we wanted seems like we could jam these into a one-liner to remove one layer of nesting
```
if syncType, exists := c.Params["syncType"]; exists {
  if ! IsValidSyncType(syncType) {
    return nil, errors.New(invalidSyncTypeMessage)
  }
}
```